segmentation/scripts/instance_segm.py

```diff
@@ -0,0 +1,281 @@
+from tqdm import tqdm
+import cv2
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def semantic_to_instance_mask(mask_rgb, target_rgb=(255, 255, 255)):
+    """
+    Convert an RGB mask to an instance segmentation mask.
+
+    Args:
+        mask_rgb (numpy.ndarray): RGB mask (H, W, 3).
+        target_rgb (tuple): Target RGB value to isolate (e.g., (255, 255, 255)).
+
+    Returns:
+        instance_mask (numpy.ndarray): Instance segmentation mask with unique labels for each instance.
+    """
+    # Create a binary mask for the target class
+    binary_mask = np.all(mask_rgb == np.array(target_rgb), axis=-1).astype(
+        np.uint8)  # Shape: (H, W)
+
+    # Initialize the instance mask
+    instance_mask = np.zeros_like(binary_mask, dtype=np.int32)
+
+    # Find contours
+    contours, _ = cv2.findContours(binary_mask, cv2.RETR_EXTERNAL,
```

Comment:
Just a random thought, might be useful to filter contours based on their length (contour consisting of 2-3 points is either a noise or a very low confidence prediction)